flow/envs/green_wave_env.py

```diff
@@ -165,9 +169,11 @@ def _apply_rl_actions(self, rl_actions):
                         self.traffic_lights.set_state(
                             node_id='center{}'.format(i),
                             state='rrrGGGrrrGGG', env=self)
+                    self.last_change[i, 0] = 0.0
                     self.last_change[i, 2] = 1
-            else:
-                if action:
+            else: # currently green
+                self.last_change[i, 0] += self.sim_step
+                if self.last_change[i, 0] >= self.min_green_time and rl_mask[i]:
```

Comment:
Why does `self.last_change` get incremented inside the `else` when it didn't before? (not too familiar with the green wave stuff though so I might just be missing something obvious)

Reply:
self.last_change used to only keep track of last change since a yellow phase began. Now it also keeps track of time since a green phase began. So now, last_change[i, 0] resets when the green phase begins (it didn't use to) and increments during the green phase as well.